github3/github.py

```diff
@@ -772,6 +774,11 @@ def login(self, username=None, password=None, token=None,
         # The Session method handles None for free.
         self.session.two_factor_auth_callback(two_factor_callback)
 
+        # If running as an App the `Accept` header needs to change.
+        if app:
```

Comment:
If instead this becomes a method on `GitHubSession` that we call, e.g., `self.session.configure_for_app()` then we don't need the parameter to the class or the wild proliferation of setting this header in every conceivable place